src/app/services/actions/query-action-creators.ts

```diff
@@ -0,0 +1,26 @@
+import { IAction } from '../../../types/query-runner';
```

Comment:
Is this a temporary location for IAction?  Why would a generic notion like IAction live in a specific query-runner file?

Reply:
Good question.

It didn't sound reasonable to create a `types/action` and only have one type in it. What do you think?

Reply:
Or we could get rid of the type IAction.